### PR TITLE
Add new indexed true color composite to generate hybrid green channels.

### PR DESCRIPTION
**Pull request by simon-sat**
This PR adds a new `true_color_indexed` composite for MTG/FCI that builds upon @strandgren's hybrid NDVI composite that's currently used in satpy `main`.
As with the current `main`, a scaling factor is used to adjust the ratio between the NIR and green channels to account for the green channel missing the chlorophyll peak. But, in contrast, this PR uses two indices to determine the scaling and is based on a physical comparison of the spectral response function compared to an ideal sensor:

```
ndvi = (nir - red) / (nir + red)
ndbi = (red - green) / (red + green)

factor = max(ndvi * veggie_scaling, ndbi * soil_scaling
```
Where `veggie_scaling` and `soil_scaling` are precomputed factors based on the spectral response function. These scaling factors attempt to compare what the satellite actually sees to the idealised spectral response of vegetation and soil as defined in NASA's [speclib](https://speclib.jpl.nasa.gov) database. The `ndbi` parameter is designed to estimate the prevalence of soil, as opposed to vegetation, within a given pixel and helps correct for some of the unpleasant green/yellow shading that can occur in barren regions when NDVI alone is used.

The reason for applying NDBI in addition to NDVI is that the reflectance spectra of soil is not constant between the original green bands of most satellite instruments and the shifted green to hit the chlorophyll peak, as shown here:
SRF
This graph is for FCI and shows the SRFs of the 4 lowest wavelength channels, plus the reflectance spectra of grass + soil and the theoretical shifted SRF to match the green peak.
There is a small increase in soil reflectance between the original and shifted green, but for soil the NDVI is low and hence this increase is not accounted for. But NDBI will be high and this enables a small increase in reflectance for cases where NDVI is low and NDBI high.

This is currently a draft PR as it only supports FCI. I intend to add ABI, AHI and AMI as well. I also intend to make available the scripts I used to calculate the scaling factors, but I may do that in a separate repository as I don't think they're appropriate to satpy's repo itself.

 - [ ] Tests added 
 - [ ] Fully documented

### CODE REVIEW

File: satpy/composites/spectral.py [hunk 1]
```diff
@@ -163,6 +172,51 @@ def __call__(self, projectables, optional_datasets=None, **attrs):
...
+        ndvi_input = self.match_data_arrays([projectables[1], projectables[2]])
+        ndbi_input = self.match_data_arrays([projectables[0], projectables[1]])
+
+        ndvi = _calc_norm_index(ndvi_input[1], ndvi_input[0])
+        ndvi.data = da.where(ndvi > self.ndvi_min, ndvi, self.ndvi_min)
+        ndvi.data = da.where(ndvi < self.ndvi_max, ndvi, self.ndvi_max)
+
+        ndbi = _calc_norm_index(ndbi_input[1], ndbi_input[0])
+        ndbi.data = da.where(ndbi > self.ndvi_min, ndbi, self.ndvi_min)
+        ndbi.data = da.where(ndbi < self.ndvi_max, ndbi, self.ndvi_max)
```
Comment (mraspaud) [inline satpy/composites/spectral.py:209]: so you compare ndbi to ndvi?

Comment (simonrp84) [reply satpy/composites/spectral.py:209]: Not on those lines, no. I just use the same thresholds as they're appropriate to both in my tests. I could make separate thresholds for each now that you bring it up - would enable greater customisation.

Comment (djhoese) [reply satpy/composites/spectral.py:209]: This is the same as clipping right? Using `np.clip` with the min and max should perform better...at least I'd hope so.

Comment (simonrp84) [reply satpy/composites/spectral.py:209]: Ok, have updated the code for that now!

**Pull request closed without merging**